Python interpreter and libraries detection

robertodr opened the pull request:
Changes to the Python detection:
*  The detection is now split into interpreter and libraries/headers. There are two modules: python_interpreter.cmake and python_libs.cmake. They mimick the behaviour of the standard CMake modules. The latter module depends on the former: the CMake admin has to take care of this.
* The python_interpreter.cmake module sets a --python option in the setup.py script. The end-user can thus set his/her own installation of the interpreter (and thus also libraries/headers) *via* command-line
* Tests were added for both the python_interpreter.cmake and the python_libs.cmake with and without explicit specification of an interpreter from the user. 
* For the python_libs.cmake test the [basic embedding example](https://docs.python.org/2/extending/embedding.html#very-high-level-embedding) from the Python documentation was copy-pasted.
* .travis.yml know installs libpython-dev prior to test execution.

Possible limitations and ideas for further work:
* The only way to specify that a working interpreter is required is to hardcoded the information **after** the project has been bootstrapped.
* Same goes for minimum required version specification. It has to be hardcoded **after** project bootstrap was done.
* The python_interpreter.cmake test cases compile a dummy C++ example program. Thus only detection is tested of the interpreter is currently tested. It might be appropriate to also try and use the interpreter.

bast commented:
Thanks Roberto. But some tests still fail and therefore I cannot accept at this moment. Let me know if you need assistance with those.

robertodr commented:
@bast I think it's a problem with the Travis configuration.

robertodr commented:
@bast closing for the moment so that you don't get flooded by failing CI jobs. I will open a new one when the build on Travis is fixed.

robertodr commented:
@bast All tests are now passing. Corrected issues:
* The test functions added to test.py were not PEP8 conforming.
* Wrong name of .deb package shipping development Python version.

I also switched the Travis-CI worker setup so that it uses the container-based workers, as described
[here](http://docs.travis-ci.com/user/migrating-from-legacy/)
Advantages:
* Builds start much faster.
* Previous setup is legacy Travis-CI infrastructure and it will, eventually, be discontinued (I suppose).

bast commented:
Big thanks for switching to the new Travis workers! There is still some issue with Python 3 (Appveyor) but
we can fix this later. One request though: can you please squash your commits into one (or few) commit(s). Every commit should be pickable by itself.

robertodr commented:
I seem not to be able to do so... Could you point out how to do it?

bast commented:
One possibility is an interactive rebase:
```
$ git rebase -i <oldest hash where you want to squash into>
```

The other (possibly simpler) is a soft reset:
```
$ git reset --soft <the hash just before your work started>
$ git status
$ git commit
```

Since you rewrote history you then need to push with `--force`.

robertodr commented:
OK, I think I have managed.

bast commented:
Thanks! I let the Appveyor complain for the moment. We can fix the Python 3 issues separately.
